### DISCUSSION TITLE
UX improvements (realtime update, batch processing, etc.)

### DISCUSSION THREAD

**Discussion by BlinkDL** [hlky/stable-diffusion#90] [Ideas]
Thank you for the wonderful GUI.

1. Currently the images are shown only when all [batchCount] x [batchSize] images are finished. It will be better if the images are dynamically shown whenever a batch is completed (that is, whenever we see new images in the folder).

2. Batch processing of a long list of prompts.

3. Img2img: paste from clipboard & one-click set crop region to full image.

4. A manager for all previous generations.

**Comment by JoshuaKimsey** [comment]
For number 1, I do agree, it would be nice to see the images pop into the webui as they are rendered in a batch. It may be this was initially avoided due to potential slow downs with inference speed, like the Gradio loading bar initially caused (an odd bug... The loading bar should never need 10% of a 3090's power to render...), but I don't think loading new images dynamically into the UI would cause that much of a slowdown. Not a significant one at least.

**Comment by micwalk** [comment]
re. 1: Realtime updates should be possible to implement after this issue is completed in gradio: https://github.com/gradio-app/gradio/issues/776
